test/agentchat/contrib/vectordb/test_mongodb.py

```diff
@@ -0,0 +1,283 @@
+    # Confirm just the 2s remain
+    clxn = db.get_collection(MONGODB_COLLECTION)
+    assert {2, "2"} == {doc["_id"] for doc in clxn.find({})}
+
+
+def test_get_docs_by_ids(db_with_indexed_clxn, example_documents):
+    db, collection = db_with_indexed_clxn
+    # Insert example documents
+    db.insert_docs(example_documents, collection_name=MONGODB_COLLECTION)
+
+    # Test without setting "include" kwarg
+    docs = db.get_docs_by_ids(ids=[2, "2"], collection_name=MONGODB_COLLECTION)
+    assert len(docs) == 2
+    assert all([doc["id"] in [2, "2"] for doc in docs])
+    assert set(docs[0].keys()) == {"id", "content", "metadata"}
+
+    # Test with include
+    docs = db.get_docs_by_ids(ids=[2], include=["content"], collection_name=MONGODB_COLLECTION)
+    assert len(docs) == 1
+    assert set(docs[0].keys()) == {"id", "content"}
+
+    # Test with empty ids list
+    docs = db.get_docs_by_ids(ids=[], include=["content"], collection_name=MONGODB_COLLECTION)
+    assert len(docs) == 0
+
+
+def test_retrieve_docs(db, example_documents):
+    # Create collection
+    db.delete_collection(MONGODB_COLLECTION)
+    collection = db.get_collection(MONGODB_COLLECTION)
+    # Sanity test. Retrieving docs before documents have been added
+    results = db.retrieve_docs(queries=["Cats"], collection_name=MONGODB_COLLECTION, n_results=2)
+    assert results == []
+    # Insert example documents
+    db.insert_docs(example_documents, collection_name=MONGODB_COLLECTION)
+
+    # Sanity test. Retrieving docs before the search index had been created
+    db.retrieve_docs(queries=["Cats"], collection_name=MONGODB_COLLECTION, n_results=2)
+    # Create the index
+    db.create_vector_search_index(collection=collection, index_name=MONGODB_INDEX)
+
+    # Begin testing Atlas Vector Search
+    # NOTE: Indexing may take some time, so we must be patient on the first query.
+    # Immediately adding documents and then querying is only standard for testing
+
+    n_results = 2  # Number of closest docs to return
+
+    success = False
+    retries = RETRIES
+    while retries and not success:
+        results = db.retrieve_docs(queries=["Cats"], collection_name=MONGODB_COLLECTION, n_results=n_results)
+        if len(results[0]) == n_results:
+            success = True
+        else:
+            retries -= 1
+            sleep(DELAY)
+    if not success:
+        raise OperationFailure(f"Failed to retrieve docs after {RETRIES} retries, waiting {DELAY} seconds after each.")
+
+    assert {doc[0]["id"] for doc in results[0]} == {1, 2}
+
+    # Empty list of queries returns empty list of results
+    results = db.retrieve_docs(queries=[], collection_name=MONGODB_COLLECTION, n_results=n_results)
+    assert results == []
+
+    # Empty list of queries returns empty list of results
+    queries = ["Some good pets", "What kind of Sandwich?"]
+    results = db.retrieve_docs(queries=queries, collection_name=MONGODB_COLLECTION, n_results=2)
```

Comment:
Please add a test with `distance_threshold` set. I think it's not working.

Reply:
added a test for distance threshold :)